### DISCUSSION TITLE
example code and data for multi-class classification problem using one-hot encode

### DISCUSSION THREAD

**Discussion by GZ82** [chemprop/chemprop#1158] [Ideas]
I find a data example for multi-class here: [mol_multiclass.csv](https://github.com/chemprop/chemprop/tree/main/examples#:~:text=mol.csv-,mol_multiclass,-.csv)
but labels are 0, 1, 2 instead of one-hot encode e.g., [1, 0, 0]
hope can provide and data examples together will code for this type classification.
because I heard that one-hot is better for multi-class classification problem.
Thank you!

**Comment by KnathanM** [comment]
Hi @GZ82, thanks for the question. 

We store the class labels instead of a one-hot encoding in the datapoints because that is the format our loss functions expect. Internally the loss functions then convert these class labels to one-hot encodings. (See [torch.nn.functional.cross_entropy](https://pytorch.org/docs/stable/generated/torch.nn.functional.cross_entropy.html) and this [line](https://github.com/chemprop/chemprop/blob/1b8f4133b092fb0241f690f93a2b11e7db07c6dd/chemprop/nn/metrics.py#L382) in our code.)

At prediction time, chemprop predicts `n_classes` values ([here](https://github.com/chemprop/chemprop/blob/1b8f4133b092fb0241f690f93a2b11e7db07c6dd/chemprop/nn/predictors.py#L310-L314)) which are then softmaxed to give probablities and the highest probability class is then converted back to a class label for output (at least in the CLI).

**Comment by KnathanM** [comment]
I forgot to add some example code. This [file](https://github.com/chemprop/chemprop/blob/main/tests/cli/test_cli_classification_mol_multiclass.py) has our tests for the CLI and this [file](https://github.com/chemprop/chemprop/blob/main/tests/integration/test_classification_mol_multiclass.py) has our tests for using chemprop as a script. I don't think we have an example notebook currently...
